src/client.ts

```diff
@@ -11,47 +11,66 @@ import { Rows } from './rows'
 import type { ClickHouseSettings } from './settings'
 
 export interface ClickHouseClientConfigOptions {
+  /** A ClickHouse instance URL. Default value: `http://localhost:8123`. */
   host?: string
+  /** The timeout to setup a connection in milliseconds. Default value: `10_000`. */
   connect_timeout?: number
+  /** The request timeout in milliseconds. Default value: `30_000`. */
   request_timeout?: number
+  /** Maximum number of sockets to allow per host. Default value: `Infinity`. */
   max_open_connections?: number
 
   compression?: {
+    /** `response: true` instructs ClickHouse server to respond with compressed response body. Default: true. */
     response?: boolean
+    /** `request: true` enabled compression on the client request body. Default: false. */
     request?: boolean
   }
-  // tls?: TlsOptions;
-
+  /** The name of the user on whose behalf requests are made. Default: 'default'. */
   username?: string
+  /** The user password. Default: ''. */
   password?: string
-
+  /** The name of the application using the nodejs client. Default: 'clickhouse-js'. */
   application?: string
+  /** Database name to use. Default value: `default`. */
   database?: string
+  /** ClickHouse settings to apply to all requests. Default value: {}  */
   clickhouse_settings?: ClickHouseSettings
   log?: {
+    /** Enable logging. Default value: false.  */
     enable?: boolean
+    /** A class to instantiate a custom logger implementation. */
     LoggerClass?: new (enabled: boolean) => Logger
   }
 }
 
 export interface BaseParams {
+  /** ClickHouse settings that can be applied on query level. */
   clickhouse_settings?: ClickHouseSettings
+  /** Parameters for query binding. https://clickhouse.com/docs/en/interfaces/http/#cli-queries-with-parameters */
   query_params?: Record<string, unknown>
+  /** AbortSignal instance to cancel a request in progress. */
```

Comment:
```suggestion
  /** AbortSignal instance (using `node-abort-controller` package) to cancel a request in progress. */
```

Reply:
Ok, added. Note that any compatible implementation does the trick.